src/NBitcoin/ChainedHeader.cs

```diff
@@ -4,6 +4,63 @@
 
 namespace NBitcoin
 {
+    /// <summary>
+    /// Represents the availability state of a block.
+    /// </summary>
+    public enum BlockDataAvailabilityState
+    {
+        /// <summary>
+        /// A <see cref="BlockHeader"/> is presented, the block data is not available.
+        /// </summary>
+        HeaderOnly,
+
+        /// <summary>
+        /// We are interested in downloading the <see cref="Block"/> that is being represented by the current <see cref="BlockHeader"/>. 
+        /// This happens when we don't have block which is represented by this header and the header is a part of a chain that
+        /// can potentially replace our consensus tip because its chain work is greater than our consensus tip's chain work.
+        /// </summary>
+        BlockRequired,
```

Comment:
so here the block is not available but the header is. Is it not like HeaderOnly then?

Reply:
the distinction is that HO we don't want to download (it is not interesting for us), while this one is
so yes, we still have only the header, but we feel differently about it, so the actual physical availability have not changed, but its context did